HTML-681: enrollInProgram tag should set the location

mogoodrich opened the pull request:
<!--- Add a pull request title above in this format -->
<!--- real example: 'HTML-671 Current Status in the Module Information page shows the binary values' -->
<!--- 'HTML-JiraIssueNumber JiraIssueTitle' -->
## Description of what I changed
<!--- Describe your changes in detail -->
<!--- It can simply be your commit message, which you must have -->


## Issue I worked on
<!--- This project only accepts pull requests related to open issues -->
<!--- Want a new feature or change? Discuss it in an issue first -->
<!--- Found a bug? Point us to the issue/or create one so we can reproduce it -->
<!--- Just add the issue number at the end: -->
see https://issues.openmrs.org/browse/HTML-

## Checklist: I completed these to help reviewers :)
<!--- Put an `x` in the box if you did the task -->
<!--- If you forgot a task please follow the instructions below -->
- [ ] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

  No? -> [read here](https://wiki.openmrs.org/display/docs/Pull+Request+Tips) on how to squash multiple commits into one

- [ ] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [ ] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [ ] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [ ] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

mogoodrich commented:
Sorry, a fair amount of formatting changes appeared in this PR, but shouldn't be too hard to find the actual changes.

mogoodrich commented:
And @mks-d  more an fyi for you... not sure if you use the enrollInProgram functionality at all

mseaton commented on api/src/main/java/org/openmrs/module/htmlformentry/FormSubmissionActions.java:
```diff
@@ -374,26 +372,31 @@ public void modifyObs(Obs existingObs, Concept concept, Object newValue, Date ne
 	 * @param program Program to enroll the patient in
 	 * @param enrollmentDate the date to enroll the patient in the program
 	 * @param states list of states to set as initial in their workflows
 	 */
```
Javadoc on locationTag parameter?  What is this for?

mseaton commented on api/src/main/java/org/openmrs/module/htmlformentry/HtmlFormEntryUtil.java:
```diff
@@ -1165,6 +1164,24 @@ public static LocationTag getLocationTag(String identifier) {
+	 * @return
+	 */
+	public static Location getFirstAncestorWithTag(Location location, LocationTag locationTag) {
+		if (location.hasTag(locationTag.getName())) {
```
Might want a not-null check on location here, just in case.

mseaton reviewed (approved):
Looks good, a few minor things to ideally address.

mogoodrich commented:
Thanks @mseaton ... will address.

Looks like I forgot to push up the new test form I added as well (which resulted in failing Travis tests).  Just corrected that.

mogoodrich commented:
Addressed PR comments, merging, thanks @mseaton !
